### PR TITLE
Some Container_types-related tidy-ups

### PR DESCRIPTION
**Pull request by mshinwell**
1. Stop making vast amounts of map and hash table modules in `Simplify_binary_primitive` that we don't use.
2. Remove the `output` functions from `Container_types` which also aren't used, and are a nuisance.

### CODE REVIEW

File: middle_end/flambda2/tests/mlexamples/digest.ml [hunk 1]
```diff
@@ -757,9 +757,7 @@ let file filename =
     d
   | exception e ->
     close_in ic;
-    raise e
-
-let output chan digest = output_string chan digest
+    raise e output_string chan digest
```
Comment (lukemaurer) [inline middle_end/flambda2/tests/mlexamples/digest.ml:761]: Was this change intended? I don't see how it relates to the commit (and it seems a bit strange; is the point just to be sure `output_string`, `chan`, and `digest` are used?)

Comment (mshinwell) [reply middle_end/flambda2/tests/mlexamples/digest.ml:761]: haha no, that is a mistake left over from `sed` that the compiler didn't catch (!)

File: middle_end/flambda2/simplify/simplify_binary_primitive.ml [hunk 2]
```diff
@@ -86,7 +86,23 @@ end = struct
...
+
+      val add : elt -> t -> t
+
+      val is_empty : t -> bool
+
+      val cardinal : t -> int
+
+      val get_singleton : t -> elt option
+
+      val elements : t -> elt list
```
Comment (Gbury) [inline middle_end/flambda2/simplify/simplify_binary_primitive.ml:104]: Why is this signature explicit ? It doesn't seem to be needed (the `Set` module is not exported, and the signature is simple enough).

**Review by Gbury — APPROVED** [review]
